studio/components/ui/PageTelemetry.tsx

```diff
@@ -1,17 +1,22 @@
 import { useParams, useTelemetryProps } from 'common'
 import { useSelectedOrganization } from 'hooks'
 import { post } from 'lib/common/fetch'
-import { API_URL, IS_PLATFORM } from 'lib/constants'
+import { API_URL, IS_PLATFORM, LOCAL_STORAGE_KEYS } from 'lib/constants'
 import { observer } from 'mobx-react-lite'
 import { useRouter } from 'next/router'
-import { FC, useEffect } from 'react'
+import { PropsWithChildren, useEffect } from 'react'
 
-const PageTelemetry: FC = ({ children }) => {
+const PageTelemetry = ({ children }: PropsWithChildren<{}>) => {
   const router = useRouter()
   const { ref } = useParams()
   const telemetryProps = useTelemetryProps()
   const selectedOrganization = useSelectedOrganization()
 
+  const consent =
+    typeof window !== 'undefined'
+      ? localStorage.getItem(LOCAL_STORAGE_KEYS.TELEMETRY_CONSENT)
+      : null
```

Comment:
```suggestion
  const [consent] = useState(() =>
    typeof window !== 'undefined'
      ? localStorage.getItem(LOCAL_STORAGE_KEYS.TELEMETRY_CONSENT)
      : null
  )
```

Reading from localStorage every render can be pretty slow, but we can cache it with a `useState`. You'll need to check that it doesn't cause any hydration mismatch errors though and also reset it if you need to.